packages/core/docs/guide/checkout.md

```diff
@@ -0,0 +1,369 @@
+<template>
+  <form @submit.prevent="save({ shippingDetails: shippingForm })">
+    <!-- form fields -->
+    <button type="submit" :disabled="loading">{{ $t('Submit') }}</button>
+  </form>
+</template>
+
+<script>
+import { useShipping } from '{INTEGRATION}';
+import { onSSR } from '@vue-storefront/core';
+
+export default {
+  setup () {
+    const {
+      save,
+      loading,
+      shipping
+    } = useShipping();
+
+    const shippingForm = ref(/* object for shipping details */);
+
+    return {
+      shippingForm,
+      save
+    };
+  }
+}
+</script>
+```
+
+## Selecting a shipping method
+
+`VsfShippingProvider` is a component that aggregates one or more shipping methods from a single provider like FedEx or DHL. This component is usually the only thing that we need to integrate a particular vendor into our project and is always delivered as third-party integration.
+
+The component is responsible for:
+- Loading and displaying available shipping methods.
+- Loading selected shipping method.
+- Selecting and configuring shipping method.
+
+All we have to do is to import a component and add it to the template.
+
+```vue
+<VsfShippingProvider
+  @submit="$router.push('/checkout/billing')"
+/>
+```
+
+`VsfShippingProvider` emits `submit` event when shipping method is selected, configured and user clicks submit button.
+
+### Extending `VsfShippingProvider` and reacting to its events
+
+We can pass asynchronous functions to the `VsfShippingProvider` component to hook into different events within its lifecycle, override initial function parameters or react to specific events like method selection. We will call these methods "hooks".
+
+Because every shipping provider is different, not all of them are present in every integration. Always refer to the documentation of a specific provider to learn which hooks are available.
+
+- **beforeLoad** `(config => config)` - Called before loading shipping methods.
+- **afterLoad** `(shippingMethodsResponse => shippingMethodsResponse.shippingMethods)` - Called after loading shipping methods.
+- **beforeSelect** `(shippingMethod => shippingMethod)` - Called before selecting shipping method.
+- **afterSelect** `(selectedShippingMethod => void)` - Called after selecting shipping method.
+- **beforeSelectedDetailsChange** `(details => details)` - Called before modifying currently picked shipping method, e.g. selecting parcel locker on the map.
+- **afterSelectedDetailsChange** `(details => void)` - Called after modifying currently picked shipping method.
+- **onError** `(({ action, error }) => void)` - Called when some operation throws an error.
+
+```vue
+<VsfShippingProvider
+  :beforeLoad="beforeLoad"
+  :afterLoad="afterLoad"
+  :beforeSelect="beforeSelect"
+  :afterSelect="afterSelect"
+  :beforeSelectedDetailsChange="beforeSelectedDetailsChange"
+  :afterSelectedDetailsChange="afterSelectedDetailsChange"
+  :onError="onError"
+/>
+```
+
+:::details Loading and saving a shipping method
```

Comment:
how about not making this as toggle and just adding a paragraph

## Accessing shipping method details outside the component

And describing for what typically this composable is used for and that it shares the state with the component and that's it? 

You don't even have to rephrase that much